bitcoin/src/blockdata/transaction.rs

```diff
@@ -1129,11 +1129,14 @@ impl InputWeightPrediction {
 
     /// Computes the **signature weight** added to a transaction by an input with this weight prediction,
     /// not counting the prevout (txid, index), sequence, potential witness flag bytes or the witness count varint.
+    /// Note that this functions return value is bounded at (U32::MAX * 2) * 4 weight units.  This is done to avoid
```

Comment:
In 61c77f7bbe9a3a6f5e205a8cd9af773c77d6d802:

`U32` should be `u32` and `functions` should be `function's`. But even with those fixes the function is not really correct. For one thing, of the two private fields, only one is multiplied by 4 so the actual theoretical limit is `u32::MAX * 5`, not `u32::MAX * 2 * 4`. But also, there are no legal transactions that simultaneously have large scriptSigs and witnessScripts, so you can't even hit this maximum.

Better to just say "This function's internal arithmetic saturates at `u32::MAX`, so the return value of this function may be inaccurate for extremely large witness predictions."